src/inmanta/data/__init__.py

```diff
@@ -735,26 +741,38 @@ def __init__(
         self,
         field_type: Type[T],
         required: bool = False,
-        unique: bool = False,
-        reference: bool = False,
+        is_many: bool = False,
         part_of_primary_key: bool = False,
+        ignore: bool = False,
+        default: object = default_unset,
         **kwargs: object,
     ) -> None:
+        """A field in a document/record in the database. This class holds the metadata one how the data layer should handle
+        the field.
+
+        :param field_type: The python type of the field. This type should work with isinstance
+        :param required: Is this value required
```

Comment:
```suggestion
        :param required: Is this value required. If true, a value can not be None.
```
Is this right? It seems to be from the implementation of the validate methods.